test/test_sparse.py

```diff
@@ -3503,6 +3503,25 @@ def test_softmax_zero_nnz(self, device, dtype):
         out = torch.sparse.softmax(t, 0)
         self.assertEqual(out.to_dense(), torch.zeros_like(t))
 
+        t = t.requires_grad_()
+
+        def fn(x):
+            return torch.sparse.softmax(x, 0).to_dense()
+        torch.autograd.gradcheck(fn, (t), check_sparse_nnz=True)
+
+    @dtypes(torch.float)
+    def test_log_softmax_zero_nnz(self, device, dtype):
+        t = torch.sparse_coo_tensor([[]], [], (3,), device=device, dtype=dtype)
```

Comment:
Are zero sizes implying zero nnz tested as well?

Reply:
There seems to be no issue with zero sizes:
```python
>>> t = torch.sparse_coo_tensor([[], []], [], (0, 3), dtype=torch.float64)
>>> t.requires_grad_()
tensor(indices=tensor([], size=(2, 0)),
       values=tensor([], size=(0,)),
       size=(0, 3), nnz=0, dtype=torch.float64, layout=torch.sparse_coo,
       requires_grad=True)
>>> torch.autograd.gradcheck(lambda x: torch.sparse.softmax(x, 0).to_dense(), t, masked=True, check_sparse_nnz=True)
True
```

Reply:
But is it tested? Looks like a great opportunity to cover degenerate cases.